packages/walletconnect-solana/src/adapter.ts

```diff
@@ -115,16 +117,52 @@ export class WalletConnectWallet {
         }
     }
 
-    async signTransaction(transaction: Transaction): Promise<Transaction> {
+    async signTransaction<T extends Transaction | VersionedTransaction>(transaction: T): Promise<T> {
         if (this._client && this._session) {
-            const { signature } = await this._client.request<{ signature: string }>({
-                chainId: this._network,
-                topic: this._session.topic,
-                request: { method: WalletConnectRPCMethods.signTransaction, params: { ...transaction } },
-            });
-            transaction.addSignature(this.publicKey, Buffer.from(base58.decode(signature)));
-
-            return transaction;
+            if (isVersionedTransaction(transaction)) {
+                const rawTransaction = Buffer.from(transaction.serialize()).toString('base64');
+                if (transaction.version === 'legacy') {
+                    const legacyTransaction = Transaction.from(transaction.serialize());
+                    const { signature } = await this._client.request<{ signature: string }>({
+                        chainId: this._network,
+                        topic: this._session.topic,
+                        request: {
+                            method: WalletConnectRPCMethods.signTransaction,
+                            params: { ...legacyTransaction, transaction: rawTransaction },
+                        },
```

Comment:
Hmm, why are the transaction properties passed as params? I realize this isn't a change from how it's currently being done, but the transaction bytes are all that should be needed, right?

Reply:
@jordansexton It was old way of doing this implemented by WalletConnect. So we are keeping for backward compatibility and introducing a new way that could support both legacy and v0 transactions.